Monika After Story/game/script-apologies.rpy

```diff
@@ -0,0 +1,260 @@
+#Create an apology db for storing our times
+#Stores the event label as a key, its corresponding data is a tuple where:
+#   [0] -> timedelta defined by: current total playtime + apology_active_expiry time
+#   [1] -> datetime.date defined by the date the apology was added + apology_overall_expiry time
+default persistent._mas_apology_time_db = {}
+
+#Create a generic apology db. We'll want to know how many times the player has apologized for mas_apology_reason
+#Allows us the ability to apply diminishing returns on affection for repeated use of the same apology
+#This db here simply stores the integer corresponding to apology reason as a key,
+#corresponding int value is the amt of times it was used
+default persistent._mas_apology_reason_use_db = {}
+
+init -10 python in mas_apology:
+    apology_db = {}
+    # Event database for apologies
+
+
+#going to need this post ev_handler init
+init python:
+    def mas_checkApologies():
+        #Let's not do extra work
+        if len(persistent._mas_apology_time_db) == 0:
+            return
+
+        #Calculate the current total playtime to compare...
+        current_total_playtime = persistent.sessions['total_playtime'] + mas_getSessionLength()
+
+        #Iter thru the stuffs in the apology time tb
+        for ev in persistent._mas_apology_time_db.keys():
```

Comment:
```suggestion
        for ev_label in persistent._mas_apology_time_db.keys():
```
since these are keys, they should have a more appropriate var name